src/Entity/Post.php

```diff
@@ -162,12 +162,12 @@ public function setContent(string $content): void
         $this->content = $content;
     }
 
-    public function getPublishedAt(): \DateTime
+    public function getPublishedAt(): \DateTimeInterface
     {
         return $this->publishedAt;
     }
 
-    public function setPublishedAt(\DateTime $publishedAt): void
+    public function setPublishedAt(\DateTimeInterface $publishedAt): void
```

Comment:
Should be `\DateTimeImmutable` to avoid `\DateTime` instances, same for `Comment::setPublishedAt()`.

Reply:
Otherwise, we can convert here to immutable date time if so:
```php
if ($publishedAt instanceof \DateTime) {
    $publishedAt = \DateTimeImmutable::createFromMutable($publishedAt);
}
```

Reply:
@yceruto, What about to create a data transformer for those properties?

Reply:
Yes, it could be so, but still I'm not sure if it's worth it.